### PR TITLE
AVX-23092: add support for Edge as a Spoke

### PR DESCRIPTION
**Pull request by shuoliugit**
This is a feature for 6.8.

### CODE REVIEW

File: aviatrix/resource_aviatrix_edge_spoke.go [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Management interface configuration. Valid values: 'DHCP' and 'Static'.",
+				ValidateFunc: validation.StringInSlice([]string{"DHCP", "Static"}, false),
+			},
+			"wan_interface_ip_prefix": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "WAN interface IP / prefix.",
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_edge_spoke.go]: IP/prefix.

File: aviatrix/resource_aviatrix_edge_spoke.go [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+			"wan_default_gateway_ip": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "WAN default gateway IP.",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"lan_interface_ip_prefix": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LAN interface IP / prefix.",
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_edge_spoke.go]: Same.

File: aviatrix/resource_aviatrix_edge_spoke.go [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+			},
+			"management_egress_ip_prefix": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Management egress gateway IP / prefix.",
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_edge_spoke.go]: Same.

File: aviatrix/resource_aviatrix_edge_spoke.go [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+				Optional:    true,
+				ForceNew:    true,
+				Default:     false,
+				Description: "Enable management over private network.",
+			},
+			"management_interface_ip_prefix": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Management interface IP / prefix.",
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_edge_spoke.go]: Same.

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_edge_spoke.go [hunk 5]
```diff
@@ -0,0 +1,722 @@
...
+				Description: "Switch to enable/disable learned CIDR approval for BGP Spoke Gateway. Valid values: true, false.",
+			},
+			"approved_learned_cidrs": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsCIDR,
+				},
+				Optional:    true,
+				Description: "Approved learned CIDRs for BGP Spoke Gateway. Available as of provider version R2.21+.",
```
Comment (mlin-aviatrix) [inline aviatrix/resource_aviatrix_edge_spoke.go]: Don't need the last part `Available as of provider version...`

**Review by mlin-aviatrix — DISMISSED** [review]

**Comment by mlin-aviatrix** [comment]
I meant to request changes instead of approve. There is one small change.

**Review by mlin-aviatrix — DISMISSED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**